### DISCUSSION TITLE
Inverter power consumption

### DISCUSSION THREAD

**Discussion by krisshurley** [wlcrs/huawei_solar#863] [General]
Is there some way to calculate the power consumption of the inverter it self?
Would love to be able to have a sensor for this.

**Comment by Roving-Ronin** [comment]
@krisshurley Why... 2 x M1 6kW use about 10-13W when no PV production. If you want to have a sensor for this, then add another (helper sensor) to convert from Power (W) to Energy (kWh) and finally utility_meter to take the energy and give you daily/weekly etc readings....

Then you need to write a bespoke sensor(s) for this in yaml and put it into the /packages directory.

If you don't have a battery, this would be a simple  calc of 'inverter_input_power - inverter_active_power', but same as if with a LUNA battery come nightime the inverter_active_power will be drawing power (from grid/battery) and there's no input_power so the sensor will be negative W (and positive during the daytime), so if not worked around, your energy sensors will be screwed up.

**Reply by krisshurley** [reply]
Okay, I thought the inverter consumed more than that.
The reason I asked is that I measure almost all other consumption in the house individually.

I have a Luna battery, but I still don’t quite understand how to calculate how much the inverter and battery draw from the grid, the panels, or the battery itself. In other words, how much the battery and inverter consume just to keep themselves running.
